From: David Rosca <nowrep@gmail.com>
Date: Fri, 17 Apr 2015 07:35:24 +0000
Subject: Fix native file dialogs from widgets QFileDialog
X-Git-Url: http://quickgit.kde.org/?p=frameworkintegration.git&a=commitdiff&h=26406a9aa01d6f4f01a3f084b51451de8385c8fc
---
Fix native file dialogs from widgets QFileDialog

There were two issues:
* File dialogs opened with exec() and without parent were
opened, but any user-interaction was blocked in a way that
no file could be selected nor the dialog closed.

* File dialogs opened with open() or show() with parent were
not opened at all.

The first issue was caused by first calling show() and then exec()
on the native dialog.
The second one simply because in the case of dialogs with parent
show() wasn't called.

This fixes it that the show() is always called and hide() is called
before exec().

This also adds unittests for file dialogs.

REVIEW: 123335
---


--- a/src/platformtheme/kdeplatformfiledialoghelper.cpp
+++ b/src/platformtheme/kdeplatformfiledialoghelper.cpp@@ -272,6 +272,7 @@
 
 void KDEPlatformFileDialogHelper::exec()
 {
+    m_dialog->hide(); // ensure dialog is not shown (exec would block input)
     m_dialog->winId(); // ensure there's a window created
     KSharedConfig::Ptr conf = KSharedConfig::openConfig();
     KWindowConfig::restoreWindowSize(m_dialog->windowHandle(), conf->group("FileDialogSize"));
@@ -296,11 +297,11 @@
 
 bool KDEPlatformFileDialogHelper::show(Qt::WindowFlags windowFlags, Qt::WindowModality windowModality, QWindow *parent)
 {
+    Q_UNUSED(parent)
     initializeDialog();
     m_dialog->setWindowFlags(windowFlags);
     m_dialog->setWindowModality(windowModality);
-    if (!parent || (parent && !parent->inherits("QWidgetWindow"))) // see #334963 and #344586 for details
-        m_dialog->show();
+    m_dialog->show();
     KSharedConfig::Ptr conf = KSharedConfig::openConfig();
     KWindowConfig::restoreWindowSize(m_dialog->windowHandle(), conf->group("FileDialogSize"));
     return true;

